pkg/deployment/collector.go

```diff
@@ -88,12 +88,6 @@ func (c *Collector) Get() *appsv1.Deployment {
 	// see https://github.com/jaegertracing/jaeger-operator/issues/334
 	sort.Strings(options)
 
-	replicaSize := c.jaeger.Spec.Collector.Replicas
-	if replicaSize == nil || *replicaSize < 0 {
```

Comment:
Since we no longer have to check on nill. We might remove the pointer and use plain int.

Reply:
Not sure I understand your comment. The Deployment's `Spec.Replica` has the same type as our `.Spec.Collector.Replicas`, and nil means "not set".

Reply:
`c.jaeger.Spec.Collector.Replicas` is a pointer. The question is whether we can change it to plain `int`

Reply:
The deployment's `.Spec.Replicas` is also a pointer. If we change our side to plain `int`, it suddenly has a value, like `0`, which is not the same as `nil`. Once we get this ambiguity, our fix in the `inventory` package won't work anymore.

https://github.com/jaegertracing/jaeger-operator/blob/009b087d8d79a492fbeaef28ed5d6fb5446de822/pkg/inventory/deployment.go#L30-L35